models/default_local_provider.go

```diff
@@ -453,46 +457,50 @@ func (l *DefaultLocalProvider) DeleteMesheryPattern(req *http.Request, patternID
 	return l.MesheryPatternPersister.DeleteMesheryPattern(id)
 }
 
-// ImportPatternFileGithub downloads a file from a repository and stores it as a pattern for the user
-func (l *DefaultLocalProvider) ImportPatternFileGithub(req *http.Request, owner, repo, path string) ([]byte, error) {
-	githubAPIURL := fmt.Sprintf("https://api.github.com/repos/%s/%s/contents/%s", owner, repo, path)
-
-	resp, err := http.Get(githubAPIURL)
+// RemotePatternFile takes in the
+func (l *DefaultLocalProvider) RemotePatternFile(req *http.Request, resourceURL, path string, save bool) ([]byte, error) {
+	parsedURL, err := url.Parse(resourceURL)
 	if err != nil {
 		return nil, err
 	}
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("file not found")
-	}
 
-	respJSON := map[string]interface{}{}
+	// Check if hostname is github
+	if parsedURL.Host == "github.com" {
+		parsedPath := strings.Split(parsedURL.Path, "/")
+		if len(parsedPath) < 3 {
+			return nil, fmt.Errorf("malformed URL: url should be of type github.com/<owner>/<repo>/[branch]")
+		}
 
-	// Decode resp into the json object
-	if err := json.NewDecoder(resp.Body).Decode(&respJSON); err != nil {
-		return nil, err
-	}
+		owner := parsedPath[1]
+		repo := parsedPath[2]
+		branch := "main"
 
-	// Get the name of the file
-	name, ok := respJSON["name"].(string)
-	if !ok {
-		return nil, fmt.Errorf("failed to get filename from github")
-	}
+		if len(parsedPath) == 4 {
+			branch = parsedPath[3]
```

Comment:
hey @utkarsh-pro 
I think should be 
```
if len(parsedPath) >= 4 {
			branch = parsedPath[len(parsedPath)-1]
		}
```
to facilitate requests that contain `blob` 
```
{
 "url": "https://github.com/layer5io/meshery/blob/master",
  "path": ".goreleaser.yml",
  "save": true
}
```

Reply:
That's not the format. Format is `https://github.com/<org>/<repo>/[branch]`. There is no need for blob.